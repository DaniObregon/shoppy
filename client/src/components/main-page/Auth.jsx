import api from "../../config/axiosConfig"
import React from "react";
import { signInWithPopup, GoogleAuthProvider, signOut } from "firebase/auth";
import { SignUpButton } from "./SignUpButton";
import { SignOutButton } from "./SignOutButton";
import { useNavigate } from "react-router-dom";
import { auth } from "../../config/firebaseConfig";
import { useDispatch, useSelector } from "react-redux";
import { login, logout } from "../../redux/slices/userSlice";
import { Flex, Text, Stack } from "@chakra-ui/react";

export const Auth = ({ onAuthSuccess }) => {
  const provider = new GoogleAuthProvider();
  const navigate = useNavigate();
  const dispatch = useDispatch();

  // Obtiene el estado del usuario desde Redux (obtiene los datos)
  const userInfo = useSelector((state) => state.userInfo);

  const signUpWithGoogle = async () => {
    try {
      const result = await signInWithPopup(auth, provider);
      const user = result.user;
      const token = await user.getIdToken();
      const userData = {
        name: user.displayName,
        email: user.email,
        role_id: 1, // Rol por defecto (client)
        token: token, // Guardamos el token aquí
      };

      //console.log("Datos que se envían al backend:", userData);

      const response = await api.post(
        "/api/auth/signup",
        JSON.stringify(userData),
        {
          headers: {
            "Content-Type": "application/json",
            Authorization: `Bearer ${token}`, // El token se envia en un header
          },
        }
      );

      //console.log("Respuesta del backend:", response.data);

      // Accede a las propiedades del objeto `user` desde la respuesta del backend
      const { name, email, role } = response.data.user;

      //console.log("Datos recibidos del backend:", { name: name, email: email, role: role });

      // Actualizar el estado global de Redux con los datos del usuario incluyendo el token
      dispatch(
        login({
          name: name,
          email: email,
          role_id: role,
          token: token,
        })
      );

      // Si la autenticación y la creación del usuario son exitosas
      if (onAuthSuccess) {
        onAuthSuccess(user); // Llamar a la función si está definida
      }
      navigate("/"); // Redirige al home después de autenticarse (o cambiar ruta)
    } catch (error) {
      console.error("Error durante el inicio de sesión:", error);
    }
  };

  const signOutUser = async () => {
    try {
      await signOut(auth);
      dispatch(logout()); // Despachar la acción de logout en Redux
      navigate("/");
    } catch (error) {
      console.error("Error al cerrar sesión:", error);
    }
  };

  return (
    <Flex align="center" justify="center" p={4}>
      {userInfo.name ? (
        <Stack direction="row" align="center" spacing={4}>
          <Text
            p={2}
            color="gray.600"
            whiteSpace="nowrap"
            maxWidth="200px"
            overflow="hidden"
            textOverflow="ellipsis"
            fontSize={"m"}
          >
            Hola, {userInfo.name.split(" ")[0]}
          </Text>
          <SignOutButton onClick={signOutUser} />
        </Stack>
      ) : (
        <SignUpButton onClick={signUpWithGoogle} />
      )}
    </Flex>
  );
};
