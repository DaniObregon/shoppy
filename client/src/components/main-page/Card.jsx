import React from "react";
import { useNavigate } from "react-router-dom";
import { chakra, Box, Flex } from "@chakra-ui/react";
import { AddToCartButton } from "./AddToCartButton";

export const Card = ({ id, brand, model, description, price, imgUrl }) => {
  const navigate = useNavigate();

  const handleNavigate = () => {
    navigate(`/product/${id}`);
  };

  return (
    <Flex
      direction="column"
      justifyContent="center"
      alignItems="center"
      w="full"
      mx="auto"
      mb={4}
      transition="transform 0.15s ease-in-out"
      _hover={{
        transform: "scale(1.015)",
        cursor: "pointer",
      }}
      onClick={handleNavigate} // Redirige cuando se hace clic en la card
    >
      <Box
        bg="gray.300"
        h={{ base: 56, md: 64 }} //Imagen m치s grande para mantener la proporci칩n
        w="full"
        rounded="lg"
        shadow="md"
        bgSize="cover"
        bgPos="center"
        style={{
          backgroundImage: `url(${imgUrl})`,
        }}
      ></Box>

      <Box
        w={{ base: "80%", md: "70%" }} // Caja de datos peque침a y centrada
        bg="white"
        _dark={{ bg: "gray.800" }}
        mt={-10} // Superpone la caja de datos en la imagen
        shadow="lg"
        rounded="lg"
        overflow="hidden"
      >
        <chakra.h3
          py={2}
          textAlign="center"
          fontWeight="bold"
          textTransform="uppercase"
          color="gray.800"
          _dark={{ color: "white" }}
          letterSpacing={1}
        >
          {brand}
        </chakra.h3>
        <chakra.p
          py={1}
          textAlign="center"
          fontWeight="medium"
          color="gray.600"
          _dark={{ color: "gray.300" }}
          fontSize="sm"
        >
          {model}
        </chakra.p>
        <chakra.p
          py={1}
          textAlign="center"
          fontWeight="medium"
          color="gray.600"
          _dark={{ color: "gray.300" }}
          fontSize="sm"
          sx={{
            overflow: "hidden",
            display: "-webkit-box",
            WebkitLineClamp: "3",
            WebkitBoxOrient: "vertical",
            wordBreak: "break-word", // Corta palabras largas autom치ticamente
            overflowWrap: "break-word", // Compatibilidad adicional
          }}
        >
          {description}
        </chakra.p>
        <Flex
          alignItems="center"
          justifyContent="space-between"
          py={2}
          px={3}
          bg="gray.200"
          _dark={{ bg: "gray.700" }}
        >
          <chakra.span
            fontWeight="bold"
            color="gray.800"
            _dark={{ color: "gray.200" }}
          >
            ${price}
          </chakra.span>
          <AddToCartButton id={id} />
        </Flex>
      </Box>
    </Flex>
  );
};
