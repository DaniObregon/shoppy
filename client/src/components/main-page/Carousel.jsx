import React, { useEffect, useState } from "react";
import { Box, Flex, Image, Text } from "@chakra-ui/react";
import { slides } from "../../data/carouselSlides";

export const Carousel = () => {
  const [currentSlide, setCurrentSlide] = useState(0);
  const [visibleSlides, setVisibleSlides] = useState(slides);
  const [isPaused, setIsPaused] = useState(false); // Estado para manejar si la transición está pausada
  const [isHovered, setIsHovered] = useState(false); // Estado para manejar si el mouse está sobre el carrusel

  const slidesCount = visibleSlides.length;
  const carouselStyle = {
    transition: "all 1.5s",
    ml: `-${currentSlide * 100}%`,
  };

  const SLIDES_INTERVAL_TIME = 4000;
  const ANIMATION_DIRECTION = "right";

  useEffect(() => {
    const prevSlide = () => {
      setCurrentSlide((s) => (s === 0 ? slidesCount - 1 : s - 1));
    };

    const nextSlide = () => {
      setCurrentSlide((s) => (s === slidesCount - 1 ? 0 : s + 1));
    };

    const automatedSlide = setInterval(() => {
      if (!isPaused) {
        ANIMATION_DIRECTION.toLowerCase() === "left"
          ? prevSlide()
          : nextSlide();
      }
    }, SLIDES_INTERVAL_TIME);

    return () => clearInterval(automatedSlide);
  }, [slidesCount, isPaused]);

  const handleMouseEnter = () => {
    setIsPaused(true);
    setIsHovered(true);
  };

  const handleMouseLeave = () => {
    setIsPaused(false);
    setIsHovered(false);
  };

  // Función para manejar el error de carga de imágenes y eliminar el slide
  const handleImageError = (index) => {
    setVisibleSlides((prevSlides) =>
      prevSlides.filter((_, slideIndex) => slideIndex !== index)
    );
  };

  return (
    <Flex
      id="carousel"
      role="region"
      aria-label="carousel"
      w="full"
      bg="#edf3f8"
      _dark={{ bg: "#3e3e3e" }}
      p={0}
      alignItems="center"
      justifyContent="center"
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      <Flex w="full" overflow="hidden">
        <Flex pos="relative" h="500px" w="full" {...carouselStyle}>
          {visibleSlides.map((slide, index) => (
            <Box
              key={`slide-${index}`}
              flex="none"
              boxSize="full"
              shadow="md"
              transition="transform 0.3s ease"
              style={{
                transform: isHovered ? "translateY(-5px)" : "translateY(0px)",
              }}
            >
              <Text
                color="white"
                fontSize="xs"
                p="8px 12px"
                pos="absolute"
                top="0"
                whiteSpace="nowrap"
              >
                {index + 1} / {slidesCount}
              </Text>
              <Image
                src={slide.img}
                alt={`Slide ${index + 1}`}
                boxSize="full"
                objectFit="cover"
                style={{
                  opacity: isHovered ? "0.8" : "1",
                  transition: "opacity 0.3s ease",
                }}
                onError={() => handleImageError(index)} // Maneja el error de carga de imagen eliminando el slide
              />
            </Box>
          ))}
        </Flex>
      </Flex>
    </Flex>
  );
};
