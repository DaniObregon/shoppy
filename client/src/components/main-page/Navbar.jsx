import React from "react";
import {
  Box,
  Flex,
  IconButton,
  Stack,
  Collapse,
  useColorModeValue,
  useDisclosure,
} from "@chakra-ui/react";
import { HamburgerIcon, CloseIcon } from "@chakra-ui/icons";
import { Auth } from "./Auth";
import { DarkModeToggle } from "./DarkModeToggle";
import { Logo } from "./Logo";
import { DesktopNav } from "./DesktopNav";
import { MobileNav } from "./MobileNav";

export const Navbar = () => {
  const { isOpen, onToggle } = useDisclosure();

  return (
    <Box mt="60px" marginBottom={1} marginTop={5}>
      <Flex
        bg={useColorModeValue("white", "gray.800")}
        color={useColorModeValue("gray.600", "white")}
        minH={"60px"}
        py={{ base: 2 }}
        px={{ base: 10 }}
        borderBottom={1}
        borderTop={1}
        borderStyle={"solid"}
        borderColor={useColorModeValue("gray.200", "gray.900")}
        align={"center"}
        justify={"space-between"}
        marginTop={8}
        marginBottom={1}
        paddingTop="9px"
        border={-10}
      >
        <Flex
          flex={{ base: 1, md: "auto" }}
          ml={{ base: -2 }}
          display={{ base: "flex", md: "none" }}
        >
          <IconButton
            onClick={onToggle}
            icon={
              isOpen ? <CloseIcon w={3} h={3} /> : <HamburgerIcon w={5} h={5} />
            }
            variant={"ghost"}
            aria-label={"Toggle Navigation"}
          />
        </Flex>
        <Flex flex={{ base: 1 }} justify={{ base: "center", md: "start" }}>
          <Logo />
          <Flex
            display={{ base: "none", md: "flex" }}
            ml={10}
            alignItems={"center"}
          >
            <DesktopNav />
          </Flex>
        </Flex>

        <Stack
          flex={{ base: 1, md: 0 }}
          justify={"flex-end"}
          direction={"row"}
          spacing={6}
          alignItems="center" // Asegura que todo estÃ© alineado verticalmente
        >
          <Auth /> {/* Este componente maneja tanto login como logout */}
          <DarkModeToggle />
        </Stack>
      </Flex>

      <Collapse in={isOpen} animateOpacity>
        <MobileNav />
      </Collapse>
    </Box>
  );
};

<DesktopNav />;
<MobileNav />;
