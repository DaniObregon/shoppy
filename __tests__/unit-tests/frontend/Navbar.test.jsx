import "@testing-library/jest-dom";
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { Navbar } from "../../../client/src/components/main-page/Navbar";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { ChakraProvider } from "@chakra-ui/react";
import { BrowserRouter as Router } from "react-router-dom";

// Mockear módulos relacionados con Firebase
jest.mock("../../../client/src/config/firebaseConfig", () => ({
  auth: require("../../../__mocks__/firebaseConfig").auth,
}));

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: jest.fn(),
}));

jest.mock("../../../client/src/assets/butterflyLogo.png", () => "");

jest.mock("../../../client/src/components/main-page/MobileNav", () => {
  return {
    MobileNav: () => <div data-testid="mobile-nav">Mobile Nav</div>,
  };
});

const mockStore = configureStore([]);

describe("Navbar component", () => {
  let store;

  // Estado inicial (usuario no autenticado)
  const initialState = {
    userInfo: { role_id: null },
  };

  // Estado cuando el usuario tiene rol 1
  const userRole1State = {
    userInfo: { role_id: 1 },
  };

  // Estado cuando el usuario tiene rol 2
  const userRole2State = {
    userInfo: { role_id: 2 },
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders Navbar with Logo, Auth, and DarkModeToggle components", () => {
    store = mockStore(userRole1State); // Simular rol 1
    render(
      <Provider store={store}>
        <ChakraProvider>
          <Router>
            <Navbar />
          </Router>
        </ChakraProvider>
      </Provider>
    );

    // Verificar si Auth está presente (ej. botón de iniciar sesión o cerrar sesión)
    expect(screen.getByText(/ingresar/i)).toBeInTheDocument();

    // Verificar si los componentes clave están presentes
    expect(
      screen.getByRole("button", { name: /toggle navigation/i })
    ).toBeInTheDocument();

    // Verificar si DarkModeToggle está presente
    expect(
      screen.getByRole("button", { name: /toggle dark mode/i })
    ).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ChakraProvider>
          <Router>
            <Navbar />
          </Router>
        </ChakraProvider>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
