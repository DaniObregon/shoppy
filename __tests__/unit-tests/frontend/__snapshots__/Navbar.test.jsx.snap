// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-sbp024"
  >
    <div
      class="css-10zp3en"
    >
      <div
        class="css-13yncan"
      >
        <button
          aria-label="Toggle Navigation"
          class="chakra-button css-15bt6gt"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-n4g5d6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="css-1ef8uzr"
      >
        <img
          alt="Logo"
          class="chakra-image css-12p8lrt"
          src=""
        />
        <div
          class="css-xt8fz0"
        >
          <div
            aria-label="desktop navigation"
            class="chakra-stack css-d9vnmu"
          >
            <div
              class="css-0"
            >
              <a
                aria-controls="popover-content-:re:"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="chakra-link css-1gbk00m"
                id="popover-trigger-:re:"
                role="link"
              >
                Categorias
              </a>
              <div
                class="chakra-popover__popper css-1qq679y"
                style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
              >
                <section
                  class="chakra-popover__content css-r1kswd"
                  id="popover-content-:re:"
                  role="tooltip"
                  style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95);"
                  tabindex="-1"
                >
                  <div
                    class="chakra-stack css-8g8ihq"
                  >
                    <a
                      class="chakra-link css-g5555e"
                      role="group"
                    >
                      <div
                        class="chakra-stack css-1igwmid"
                      >
                        <div
                          class="css-0"
                        >
                          <p
                            class="chakra-text css-sc37ud"
                          >
                            Tecnologia
                          </p>
                          <p
                            class="chakra-text css-itvw0n"
                          />
                        </div>
                        <div
                          class="css-2gz105"
                        >
                          <svg
                            class="chakra-icon chakra-icon css-1ossod7"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                              fill="currentColor"
                            />
                          </svg>
                        </div>
                      </div>
                    </a>
                    <a
                      class="chakra-link css-g5555e"
                      role="group"
                    >
                      <div
                        class="chakra-stack css-1igwmid"
                      >
                        <div
                          class="css-0"
                        >
                          <p
                            class="chakra-text css-sc37ud"
                          >
                            Electro
                          </p>
                          <p
                            class="chakra-text css-itvw0n"
                          />
                        </div>
                        <div
                          class="css-2gz105"
                        >
                          <svg
                            class="chakra-icon chakra-icon css-1ossod7"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                              fill="currentColor"
                            />
                          </svg>
                        </div>
                      </div>
                    </a>
                    <a
                      class="chakra-link css-g5555e"
                      role="group"
                    >
                      <div
                        class="chakra-stack css-1igwmid"
                      >
                        <div
                          class="css-0"
                        >
                          <p
                            class="chakra-text css-sc37ud"
                          >
                            Herramientas
                          </p>
                          <p
                            class="chakra-text css-itvw0n"
                          />
                        </div>
                        <div
                          class="css-2gz105"
                        >
                          <svg
                            class="chakra-icon chakra-icon css-1ossod7"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                              fill="currentColor"
                            />
                          </svg>
                        </div>
                      </div>
                    </a>
                    <a
                      class="chakra-link css-g5555e"
                      role="group"
                    >
                      <div
                        class="chakra-stack css-1igwmid"
                      >
                        <div
                          class="css-0"
                        >
                          <p
                            class="chakra-text css-sc37ud"
                          >
                            Juguetes
                          </p>
                          <p
                            class="chakra-text css-itvw0n"
                          />
                        </div>
                        <div
                          class="css-2gz105"
                        >
                          <svg
                            class="chakra-icon chakra-icon css-1ossod7"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                              fill="currentColor"
                            />
                          </svg>
                        </div>
                      </div>
                    </a>
                    <a
                      class="chakra-link css-g5555e"
                      role="group"
                    >
                      <div
                        class="chakra-stack css-1igwmid"
                      >
                        <div
                          class="css-0"
                        >
                          <p
                            class="chakra-text css-sc37ud"
                          >
                            Jardin
                          </p>
                          <p
                            class="chakra-text css-itvw0n"
                          />
                        </div>
                        <div
                          class="css-2gz105"
                        >
                          <svg
                            class="chakra-icon chakra-icon css-1ossod7"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                              fill="currentColor"
                            />
                          </svg>
                        </div>
                      </div>
                    </a>
                    <a
                      class="chakra-link css-g5555e"
                      role="group"
                    >
                      <div
                        class="chakra-stack css-1igwmid"
                      >
                        <div
                          class="css-0"
                        >
                          <p
                            class="chakra-text css-sc37ud"
                          >
                            Zapatillas
                          </p>
                          <p
                            class="chakra-text css-itvw0n"
                          />
                        </div>
                        <div
                          class="css-2gz105"
                        >
                          <svg
                            class="chakra-icon chakra-icon css-1ossod7"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                              fill="currentColor"
                            />
                          </svg>
                        </div>
                      </div>
                    </a>
                    <a
                      class="chakra-link css-g5555e"
                      role="group"
                    >
                      <div
                        class="chakra-stack css-1igwmid"
                      >
                        <div
                          class="css-0"
                        >
                          <p
                            class="chakra-text css-sc37ud"
                          >
                            OFERTAS!
                          </p>
                          <p
                            class="chakra-text css-itvw0n"
                          />
                        </div>
                        <div
                          class="css-2gz105"
                        >
                          <svg
                            class="chakra-icon chakra-icon css-1ossod7"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                              fill="currentColor"
                            />
                          </svg>
                        </div>
                      </div>
                    </a>
                  </div>
                </section>
              </div>
            </div>
            <div
              class="css-0"
            >
              <a
                aria-controls="popover-content-:rg:"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="chakra-link css-1gbk00m"
                id="popover-trigger-:rg:"
                role="link"
              >
                Todos los productos
              </a>
            </div>
            <div
              class="css-0"
            >
              <a
                aria-controls="popover-content-:ri:"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="chakra-link css-1gbk00m"
                id="popover-trigger-:ri:"
                role="link"
              >
                ¿Donde estamos?
              </a>
            </div>
            <div
              class="css-0"
            >
              <a
                aria-controls="popover-content-:rk:"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="chakra-link css-1gbk00m"
                id="popover-trigger-:rk:"
                role="link"
              >
                Quienes somos...
              </a>
            </div>
            <div
              class="css-0"
            >
              <a
                aria-controls="popover-content-:rm:"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="chakra-link css-1gbk00m"
                id="popover-trigger-:rm:"
                role="link"
              >
                Contactanos
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-es01n6"
      >
        <div
          class="css-177jhv5"
        >
          <button
            class="chakra-button css-1e2mvam"
            type="button"
          >
            Ingresar
          </button>
        </div>
        <button
          aria-label="Toggle Dark Mode"
          class="chakra-button css-lt0n4j"
          data-testid="dark-mode-toggle"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21.4,13.7C20.6,13.9,19.8,14,19,14c-5,0-9-4-9-9c0-0.8,0.1-1.6,0.3-2.4c0.1-0.3,0-0.7-0.3-1 c-0.3-0.3-0.6-0.4-1-0.3C4.3,2.7,1,7.1,1,12c0,6.1,4.9,11,11,11c4.9,0,9.3-3.3,10.6-8.1c0.1-0.3,0-0.7-0.3-1 C22.1,13.7,21.7,13.6,21.4,13.7z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="chakra-collapse"
      style="overflow: hidden; display: none; opacity: 0; height: 0px;"
    >
      <div
        data-testid="mobile-nav"
      >
        Mobile Nav
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
